### PR TITLE
test: Remove unused try-block in assert_debug_log

### PR DESCRIPTION
**Pull request by maflcko**
This try block has accidentally been added by me in fa3e9f7627784ee00980590e5bf044a0e1249999.
It was unused all the time, but commit 6011c9d72d1df5c2cd09de6f85c21eb4f7eb1ba8 added a `return` in the finally block, muting all exceptions.

This can be tested by adding an `assert False` after any `with ...assert_debug_log...:` line.

### CODE REVIEW

**Review by ryanofsky — APPROVED** [review]
utACK fae91a09c453a9a95c382df765bd71e54698d5b2. I didn't know returning inside a `finally` block would cancel pending exceptions or return values, but I guess this makes sense and is a good thing to be aware of.

I'd maybe rephrase "Remove incorrect and unused try-block". I think the `try` block was fine and even potentially useful, but the `return` statement was wrong. It should be fine to remove the `try` block, though. Only difference should be that if a test is expecting an exception _and_ a log message it will now need to catch the exception inside the `assert_debug_log` block instead of being able to catch it outside.

**Comment by MarcoFalke** [comment]
> I think the try block was fine and even potentially useful, but the return statement was wrong

I think either of them is fine on their own, but in combination, at least one of them is incorrect. Though, I am happy to change to something else, if you have any suggestions.

**Comment by ryanofsky** [comment]
It's fine, I only meant to suggest changing "Remove incorrect and unused try-block" to "Remove unused try-block." AFAICT using a try block is correct and appropriate here, but nothing requires a try block right now, so it's ok to simplify and remove. But returning in a final block ignores exceptions, so it's definitely a bug to use a `return` in a final block if you _don't_ want to ignore exceptions.

**Comment by MarcoFalke** [comment]
Renamed GitHub title

**Comment by laanwj** [comment]
ACK fae91a09c453a9a95c382df765bd71e54698d5b2
